storage/postgres/storage.go

```diff
@@ -47,17 +50,72 @@ func (storage *postgresStorage) Broker() storage.Broker {
 	return &brokerStorage{storage.db}
 }
 
+func (storage *postgresStorage) Platform() storage.Platform {
+	if storage.db == nil {
+		panic("Storage is not yet Open")
+	}
+	return &platformStorage{storage.db}
+}
+
 func (storage *postgresStorage) Open(uri string) error {
 	var err error
 	if uri == "" {
 		return fmt.Errorf("storage URI cannot be empty")
 	}
 	storage.once.Do(func() {
-		storage.db, err = sqlx.Open(Storage, uri)
+		storage.db, err = sqlx.Connect(Storage, uri)
+		if err != nil {
+			logrus.Panicln("Could not connect to PostgreSQL:", err)
+		}
+
+		logrus.Debug("Updating database schema")
+		if err := updateSchema(storage.db); err != nil {
+			logrus.Panicln("Could not update database schema:", err)
+		}
+
 	})
 	return err
 }
 
 func (storage *postgresStorage) Close() error {
 	return storage.db.Close()
 }
+
+func updateSchema(db *sqlx.DB) error {
+	driver, err := migratepg.WithInstance(db.DB, &migratepg.Config{})
+	if err != nil {
+		return err
+	}
+	m, err := migrate.NewWithDatabaseInstance("file://storage/postgres/migrations", "postgres", driver)
```

Comment:
this file loading doesn't work if you import the service manager in a platform-specific project

Reply:
What do you mean by "platform-specific project"? I am not sure we are on the same page here.
I expect SM to be deployed on different platforms (CF, K8S) only by changing the configuration via env vars.

Specifically for migrations I don't expect to use anything else than files. SM should package these migrations.

Reply:
By platform-specific project I mean "service-manager-cf" for example. It will have service-manager as dependency and the only thing that it will do is override some environment variables.
Then the service-manager-cf will be deployed on CF. So having the service-manager as dependency introduces this issue where files are not looked for in the dependency but in the wrapper